web/source/osk/oskManager.ts

```diff
@@ -780,15 +894,6 @@ namespace com.keyman.osk {
       // TODO:  Move this into the VisualKeyboard class!
       // The following code will always be executed except for externally created OSK such as EuroLatin
       if(this.vkbd) {
-        // Always adjust screen height if iPhone or iPod, to take account of viewport changes
-        // Do NOT condition upon form-factor; this line prevents a bug with displaying
-        // the predictive-text banner on the initial keyboard load.  (Issue #2907)
-        if(device.touchable && device.OS == 'iOS') {
-          this.vkbd.refreshLayout(this.getKeyboardHeight());
-
-          var b: HTMLElement = this._Box, bs=b.style;
```

Comment:
Why _call_ layout code if this can be part _of_ the layout code itself?  It certainly _relates_ to layout.